spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/service/impl/DefaultTaskExecutionService.java

```diff
@@ -246,11 +270,202 @@ public long executeTask(String taskName, Map<String, String> taskDeploymentPrope
+	 * @param appDeploymentRequest the details about the deployment to be executed
+	 * @return {@code TaskManifest}
+	 */
+	private TaskManifest createTaskManifest(String platformName, TaskExecutionInformation taskExecutionInformation, AppDeploymentRequest appDeploymentRequest) {
+		TaskManifest taskManifest = new TaskManifest();
+		taskManifest.setPlatformName(platformName);
+		String composedTaskDsl = taskExecutionInformation.getTaskDefinition().getProperties().get("graph");
+		if (StringUtils.hasText(composedTaskDsl)) {
+			taskManifest.setTaskDeploymentRequest(appDeploymentRequest);
+			List<AppDeploymentRequest> subTaskAppDeploymentRequests = this.taskExecutionInfoService.createRequests(
+					taskExecutionInformation.getTaskDefinition().getTaskName(),
+					taskExecutionInformation.getTaskDefinition().getDslText());
+			taskManifest.setSubTaskDeploymentRequests(subTaskAppDeploymentRequests);
+		}
+		else {
+			taskManifest.setTaskDeploymentRequest(appDeploymentRequest);
+		}
+		return taskManifest;
+	}
+
+	/**
+	 * Return the platform specified.  If none have been specified, then "default" will be returned.
+	 *
+	 * @param taskDeploymentProperties properties to interrogate if a platform has been specified
+	 * @return name of the platform configuration to execute the task on
+	 */
+	private String getPlatform(Map<String, String> taskDeploymentProperties) {
+		String platformName = taskDeploymentProperties.get(TASK_PLATFORM_NAME);
+
+		// If not given, use 'default'
+		if (!StringUtils.hasText(platformName)) {
+			platformName = "default";
+		}
+
+		// In case we have exactly one launcher, we override to that
+		List<String> launcherNames = StreamSupport.stream(launcherRepository.findAll().spliterator(), false)
+				.map(Launcher::getName).collect(Collectors.toList());
+
+		if (launcherNames.size() == 1) {
+			platformName = launcherNames.get(0);
+		}
+
+		return platformName;
+	}
+
+	/**
+	 * Determines if the requested deployment is the same (thereby not needing an upgrade) vs different and needing to
+	 * be upgraded via the data in the manifests.  Specifically, the URI of the {@code Resource}, the app properties
+	 * and the deployment properties are all evaluated in this comparison.
+	 *
+	 * @param previousManifest the manifest for the last task execution
+	 * @param newManifest the manifest for the task execution currently being requested
+	 * @return {@code true} if no upgrade is required, {@code false} if an upgrade is required.
+	 */
+	private boolean isAppDeploymentSame(TaskManifest previousManifest, TaskManifest newManifest) {
+		boolean same;
+
+		if(previousManifest == null) {
+			return true;
+		}
+
+		Resource previousResource = previousManifest.getTaskDeploymentRequest().getResource();
+		Resource newResource = newManifest.getTaskDeploymentRequest().getResource();
+
+		try {
+			logger.debug("Previous resource was %s and new resource is %s", previousResource.getURI().toString(), newResource.getURI().toString());
```

Comment:
Catching within a log statement like this is awkward. What about if it's something else than IOException? Maybe use a proper assert for what is needed.

Reply:
The exception here is the attempt to get a URI causes the exception.  I agree it's ugly but I think the log message is valuable.  I'm open to other options as to how to address it.

Reply:
I think it's cleaner if you just just do a single debug log statetement like `logger.debug("xxx {}, xxx {}", previousResource, newResource)`. Resource toString is probably enough and then debug don't do anything if info is enabled. You could do this after equal check so that you can log that boolean as well.